src/lib/krad/packet.c

```diff
@@ -591,51 +628,63 @@ krad_packet_decode_response(krb5_context ctx, const char *secret,
                             krad_packet **rsppkt)
 {
     uchar auth[AUTH_FIELD_SIZE];
+    uint16_t len;
+    krb5_data validbuf;
     const krad_packet *req = NULL;
-    krad_packet *rsp = NULL;
+    krb5_boolean matched = FALSE;
     krb5_error_code retval;
 
-    retval = decode_packet(ctx, secret, buffer, &rsp);
-    if (retval != 0)
+    if (!cb)
+        return EINVAL;
```

Comment:
I removed the possibility to call `krad_packet_decode_response()` with a NULL pending requests iterator, because finding the matching request is needed in order to verify the Message-Authenticator.

Reply:
Actually we should still accept it in case the secret is an empty string.